src/recordpage.cpp

```diff
@@ -834,6 +856,46 @@ void RecordPage::setError(const QString& message)
     ui->applicationRecordErrorMessage->setVisible(!message.isEmpty());
 }
 
+void RecordPage::addKonsoleWidget()
+{
+    KService::Ptr service = KService::serviceByDesktopName(QStringLiteral("konsolepart"));
+
+    if (!service) {
+        return;
+    }
+
+    const auto tail = QStandardPaths::findExecutable(QStringLiteral("tail"));
+
+    if (tail.isEmpty()) {
+        return;
+    }
+
+    if (m_konsolePart) {
+        ui->recordOutputBox->layout()->removeWidget(m_konsolePart->widget());
+        m_konsolePart->deleteLater();
+    }
+    m_konsolePart = service->createInstance<KParts::ReadOnlyPart>(this);
+
+    if (!m_konsolePart) {
+        return;
+    }
+
+    m_konsoleFile = new QTemporaryFile(this);
+    m_konsoleFile->open();
+
+    const auto terminalInterface = qobject_cast<TerminalInterface*>(m_konsolePart);
+    if (!terminalInterface) {
+        qWarning("konsole kpart doesn't implement terminal interface");
+        delete m_konsolePart;
+        m_konsolePart = nullptr;
+        return;
+    }
+
+    terminalInterface->startProgram(tail, {tail, QStringLiteral("-f"), m_konsoleFile->fileName()});
+    qobject_cast<QVBoxLayout*>(ui->recordOutputBox->layout())->insertWidget(1, m_konsolePart->widget());
```

Comment:
set the `layoutName` property of the `recordOutputBox` in `designer` to something stable, e.g. `recordOutputBoxLayout`. Then use it here directly without casting